### ISSUE TITLE
requesting lots of packages with a subset of fields still causes changelogs, filelists, requires, etc.. to be loaded, resulting in slow requests

### ISSUE THREAD

**Issue by fao89** [pulp/pulpcore#2138]
Author: @jlsherrill (jsherril@redhat.com)

Redmine Issue: 9655, https://pulp.plan.io/issues/9655

---

When we make a request like this:

~~~
/pulp/api/v3/content/rpm/packages/?fields=pulp_href%2Cname%2Cversion%2Crelease%2Carch%2Cepoch%2Csummary%2Cis_modular%2Crpm_sourcerpm%2Clocation_href%2CpkgId&limit=1&offset=1000&repository_version=%2Fpulp%2Fapi%2Fv3%2Frepositories%2Frpm%2Frpm%2Fc6b93206-22bb-4c58-ba07-828c32326330%2Fversions%2F1%2F
~~~

where some subset of fields is requested, we're still seeing all of the fields being loaded from the database.  The result is that the query takes much much longer than it should

Testing on the rhel7 repo, it took about ~100 seconds on my machine to fetch all 32K packages specifying the fields above.  If I comment out the filelists, changelogs, provides/requires from the model and serializers, that drops down to ~33 seconds.  This is a huge improvement.

Reading this:  https://stackoverflow.com/questions/53319787/how-can-i-select-specific-fields-in-django-rest-framework

it looks like its may be possible to exclude certain fields from the query itself fairly easily.

**Comment by dralley** [comment]
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2121288

**Comment by dralley** [comment]
This is potentially responsible for both memory usage and timeout issues

**Comment by dralley** [comment]
TL;DR Katello does indexing so they are a heavy user of the content APIs.  If the content metadata is very... voluminous, then it can result in memory issues.  This might be a Django query cache issue, but aside from that, it would be helpful to allow avoiding querying unneeded data to begin with.
